app/src/main/java/com/github/se/signify/ui/screens/home/QuestScreen.kt

```diff
@@ -67,46 +62,36 @@ fun QuestScreen(
   LaunchedEffect(userSession.getUserId()) { userViewModel.checkAndUnlockNextQuest() }
 
   val unlockedQuests by userViewModel.unlockedQuests.collectAsState()
-
-  Scaffold(
-      modifier = Modifier.fillMaxSize().testTag("QuestScreen"),
-  ) { padding ->
-    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
-      Row(
-          verticalAlignment = Alignment.CenterVertically,
-          modifier =
-              Modifier.fillMaxWidth()
-                  .background(MaterialTheme.colorScheme.background)
-                  .padding(horizontal = 16.dp, vertical = 8.dp)) {
-            IconButton(onClick = { navigationActions.goBack() }) {
-              Icon(
-                  imageVector = Icons.AutoMirrored.Filled.ArrowBack,
-                  contentDescription = "Back",
-                  tint = MaterialTheme.colorScheme.primary)
-            }
-            Spacer(modifier = Modifier.width(20.dp))
-            Text(
-                text = stringResource(R.string.quest_screen_title),
-                fontWeight = FontWeight.Bold,
-                fontSize = 25.sp,
-                color = MaterialTheme.colorScheme.primary)
-          }
-
-      LazyColumn(
-          contentPadding = PaddingValues(vertical = 8.dp),
-          modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp).padding(padding)) {
-            items(quests.value.size) { index ->
-              val isUnlocked = index < unlockedQuests.toInt()
-              QuestBox(quest = quests.value[index], isUnlocked)
+  AnnexScreenScaffold(navigationActions = navigationActions, testTag = "QuestScreen") {
+    LazyColumn(
+        modifier = Modifier.weight(1f),
+    ) {
+      item {
+        Row(
```

Comment:
This can be replaced by a composable function named QuizTitle(){...} for better modularity and maintenance.

Reply:
"Nice suggestion, but just a small note—it’s *Quest*, not *Quiz*😄"